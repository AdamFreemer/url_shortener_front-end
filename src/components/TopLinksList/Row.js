import React from "react"

function Row(props) {
  function itemLengthLimiter(item) {
    if (!item) { return '' }
    if (item.length > 75) {
      return `${item.substring(0,75)}...`
    } else {
      return item
    }
  }

  function formatViews(views) {
    if ( views === 0 ) {
      return '-'
    } else {
      return views
    }
  }

	return (  
    <tr>
      <td className="text-right">{formatViews(props.row.views)}</td>
      <td>
        <b>{itemLengthLimiter(props.row.title)}</b><br/>
        <i><a href={props.row.url} target="_blank" rel="noopener noreferrer">{itemLengthLimiter(props.row.url)}</a></i>
      </td>
      <td>
        <a href={props.row.short_url} target="_blank" rel="noopener noreferrer">{props.row.short_url}</a>
      </td>
    </tr>
	)
}

export default Row;
