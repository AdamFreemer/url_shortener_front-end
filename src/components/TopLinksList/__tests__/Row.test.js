import React from 'react';
import renderer from 'react-test-renderer';
import Row from '../Row'

const propsMock = {
  row: {
    views: 1,
    title: "The quick brown fox jumped over the lazy dog",
    short_url: "http://www.test.com/abc",
    url: "http://www.apple.com"
  }

}

it('renders correctly a short url', () => {
  const component = renderer.create(<Row key={1} row={propsMock} />).toJSON();
  expect(component).toMatchSnapshot();
});

