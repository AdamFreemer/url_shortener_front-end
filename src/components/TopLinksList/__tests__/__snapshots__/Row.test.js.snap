// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly a short url 1`] = `
<tr>
  <td
    className="text-right"
  />
  <td>
    <b>
      
    </b>
    <br />
    <i>
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        
      </a>
    </i>
  </td>
  <td>
    <a
      rel="noopener noreferrer"
      target="_blank"
    />
  </td>
</tr>
`;
